File: src/shared/streamcmd/outputMode.go
package streamcmd

import "Theatrum/domain/models"

// OutputMode determines the muxer used by FFmpeg
type OutputMode int

const (
	OutputModeHLS  OutputMode = iota // HLS-only (ts segments, quality subdirs)
	OutputModeDASH                   // DASH-only (m4s segments, flat layout)
	OutputModeDual                   // Dual: DASH muxer with -hls_playlist 1 (both MPD and M3U8, m4s segments)
)

// DetermineOutputMode returns the appropriate output mode from a Distribution config.
func DetermineOutputMode(dist models.Distribution) OutputMode {
	if dist.HlsEnabled() && dist.DashEnabled() {
		return OutputModeDual
	}
	if dist.DashEnabled() {
		return OutputModeDASH
	}
	return OutputModeHLS
}

File: src/main/java/com/fjourdren/theatrum/infrastructure/ffmpeg/OutputMode.java
package com.fjourdren.theatrum.infrastructure.ffmpeg;

import com.fjourdren.theatrum.domain.model.Distribution;

/** The muxer FFmpeg uses for a stream. */
public enum OutputMode {

    /** HLS-only: mpegts segments, quality subdirs. */
    HLS,

    /** DASH-only: m4s segments, flat layout. */
    DASH,

    /** DASH muxer with {@code -hls_playlist 1}: both MPD and M3U8 over shared m4s segments. */
    DUAL;

    public static OutputMode determine(Distribution dist) {
        if (dist.isDualMode()) {
            return DUAL;
        }
        return dist.dashEnabled() ? DASH : HLS;
    }
}
